lib/jxl/splines.cc

```diff
@@ -159,8 +159,10 @@ void DrawFromPoints(
     }
     const float sigma =
         ContinuousIDCT(spline.sigma_dct, (32 - 1) * progress_along_arc);
-    DrawGaussian(opsin, opsin_rect, image_rect, point, multiplier, color, sigma,
-                 xs, ys, local_intensity_storage);
+    if (sigma != 0.0f) {
```

Comment:
Likely the float calculations become numerically unstable also for values very close to 0, which may or may not be compensated by the fact that what is drawn has near-zero intensity.

I would feel more comfortable by something like `abs(sigma) > .0.0000001f` in the condition here (if that fails, it should be so faint that it's invisible anyway), to not just protect against the division-by-zero but also against potential numerical instability when things are getting close to the precision limits of floats.

@sboukortt WDYT?

Reply:
Right, that would make sense, as we likely don’t want to splat NaNs all over. But then I wonder if perhaps we wouldn’t want such a protection to be more downstream where it’s more likely to be accurate? For example doing all the computations normally and then only replacing pixel values with the result of the addition if the latter is sensible.

Reply:
What is sensible though? Not inf/NaN? If sigma gets really close to 0, dividing by sigma might give huge values, I'm not sure if that would result in sensible outputs. It might help to make a test image with a long spline that goes smoothly from large sigma to negative large sigma and see what the different options produce.

@sboukortt  could you add splines to `jxl_from_tree`? Just manually entering all the dct coeffs for colors/sigma is fine (`Spline [128 numbers] x0 y0 x1 y1 ... EndSpline` or some other easy to parse syntax), we don't need something user friendly, just something we can make art and test bitstreams with :)

Reply:
> What is sensible though? Not inf/NaN?

Yes, probably something like that.

> could you add splines to `jxl_from_tree`?

That sounds helpful, I’ll try to look into this. Thanks :)

Reply:
Done in #247 if you would like to have a look.

Reply:
Thank you all for the feedback. Would `isfinite` checks within `DrawGaussian` itself suffice? In addition to `inv_sigma`, these could also apply to `sigma` and `intensity`, as if either of these values are NaN or infinite then the rest of the Gaussian calculations will result in NaN or infinite values.

Reply:
I've updated this PR to move the `sigma` and `intensity` sanity check into `DrawGaussian`.